src/main/java/core/basesyntax/service/impl/StrategyOperationServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.OperationType;
+import core.basesyntax.service.StrategyOperationService;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class StrategyOperationServiceImpl implements StrategyOperationService {
+    private Map<OperationType, OperationHandler> operationHandlerMap;
+
+    public StrategyOperationServiceImpl(Map<OperationType, OperationHandler> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler get(OperationType operationType) {
+        return operationHandlerMap.get(operationType);
```

Comment:
Consider adding a check to handle cases where `operationType` is not present in the `operationHandlerMap`. This will prevent potential `NullPointerException` if an unregistered operation type is requested.